commit 34f166a74996c0806917510656a49be1c043c7ae
Author: Rinat Ibragimov <ibragimovrinat@mail.ru>
Date:   Wed Apr 7 02:56:12 2021 +0300

    Check if x11_dpy is initialized
    
    When VA display handle is created by vaGetDisplayDRM(), x11_dpy will be
    NULL. Unchecked NULL dereference may cause a process to crash.

diff --git a/src/vdpau_driver.c b/src/vdpau_driver.c
index 0f19532..2b55b68 100644
--- a/src/vdpau_driver.c
+++ b/src/vdpau_driver.c
@@ -184,6 +184,11 @@ vdpau_common_Terminate(vdpau_driver_data_t *driver_data)
 static VAStatus
 vdpau_common_Initialize(vdpau_driver_data_t *driver_data)
 {
+    if (!driver_data->x11_dpy) {
+        /* vaGetDisplayDRM() doesn't fill ->x11_dpy */
+        return VA_STATUS_ERROR_UNKNOWN;
+    }
+
     /* Create a dedicated X11 display for VDPAU purposes */
     const char * const x11_dpy_name = XDisplayString(driver_data->x11_dpy);
     driver_data->vdp_dpy = XOpenDisplay(x11_dpy_name);
